tests.foreman.api.test_host.HostTestCase.test_positive_create_with_inherited_params does not count with global params

This test expect there are no global parameters, but there are (or there can be). This is what is in the `host.all_parameters` right before the asserts when I have global parameter `MyGlobalParameter`:

```
[{'created_at': '2018-07-31 10:32:41 UTC',
  'id': 15,
  'name': 'vxgdBHX',
  'priority': 20,
  'updated_at': '2018-07-31 10:32:41 UTC',
  'value': 'ꄖ딈胿阪𐤆㷓鰍轒亐럚浮'},
 {'created_at': '2018-07-31 10:32:40 UTC',
  'id': 14,
  'name': 'tbjSXm',
  'priority': 10,
  'updated_at': '2018-07-31 10:32:40 UTC',
  'value': '𞡭딾𗘫䶱䘤𗀰顁𝜪デ䊢𘪲갥湵鶶ꐪ𛱢洪柆ᯅ뒍糧㧭鮾佼'},
 {'created_at': '2018-07-27 09:31:05 UTC',
  'id': 3,
  'name': 'MyGlobalParameter',
  'priority': 0,
  'updated_at': '2018-07-27 09:34:54 UTC',
  'value': 'global hello'}]
```

Stacktrace:

```
self = <tests.foreman.api.test_host.HostTestCase testMethod=test_positive_create_with_inherited_params>

    @tier2
    def test_positive_create_with_inherited_params(self):
        """Create a new Host in organization and location with parameters
    
            :BZ: 1287223
    
            :id: 5e17e968-7fe2-4e5b-90ca-ae66f4e5307a
    
            :customerscenario: true
    
            :expectedresults: Host has inherited parameters from organization and
                location
    
            :CaseImportance: High
            """
        org = entities.Organization().create()
        org_param = entities.Parameter(organization=org).create()
        loc = entities.Location().create()
        loc_param = entities.Parameter(location=loc).create()
        host = entities.Host(
            location=loc,
            organization=org,
        ).create()
>       self.assertEqual(len(host.all_parameters), 2)
E       AssertionError: 4 != 2

tests/foreman/api/test_host.py:301: AssertionError
```

--- a/tests/foreman/api/test_host.py
+++ b/tests/foreman/api/test_host.py
@@ -286,7 +286,7 @@ class HostTestCase(APITestCase):
         :customerscenario: true
 
         :expectedresults: Host has inherited parameters from organization and
-            location
+            location as well as global parameters
 
         :CaseImportance: High
         """
@@ -298,10 +298,25 @@ class HostTestCase(APITestCase):
             location=loc,
             organization=org,
         ).create()
-        self.assertEqual(len(host.all_parameters), 2)
+        # get global parameters
+        glob_param_list = {(param.name, param.value) for param
+                           in entities.CommonParameter().search()}
+        # if there are no global parameters, create one
+        if len(glob_param_list) == 0:
+            param_name = gen_string('alpha')
+            param_global_value = gen_string('numeric')
+            entities.CommonParameter(
+                name=param_name,
+                value=param_global_value
+            ).create()
+            glob_param_list = {(param.name, param.value) for param
+                               in entities.CommonParameter().search()}
+        self.assertEqual(len(host.all_parameters), 2 + len(glob_param_list))
+        innerited_params = {(org_param.name, org_param.value),
+                            (loc_param.name, loc_param.value)}
+        expected_params = innerited_params.union(glob_param_list)
         self.assertEqual(
-            {(org_param.name, org_param.value),
-             (loc_param.name, loc_param.value)},
+            expected_params,
             {(param['name'], param['value']) for param in host.all_parameters}
         )
 

